File: scripts/measure/samples/add.py
"""Пример для замера: сложение, деление и накопление."""
import sys

N = int(sys.argv[1]) if len(sys.argv) > 1 else 20000


def add(a, b):
    return a + b


def div(a, b):
    return a / b


def main():
    total = 0
    for i in range(N):
        total = add(total, i)
    try:
        div(1, 0)
    except ZeroDivisionError:
        pass
    return total


print(main())

File: scripts/measure/samples/add.js
const N = Number(process.argv[2] || 20000);

function add(a, b) {
  return a + b;
}

function div(a, b) {
  if (b === 0) throw new RangeError("деление на ноль");
  return a / b;
}

function main() {
  let total = 0;
  for (let i = 0; i < N; i++) {
    total = add(total, i);
  }
  try {
    div(1, 0);
  } catch (e) {
    // проглочено намеренно
  }
  return total;
}

console.log(main());
